Set document properties into PDF Trailer

rhodgkins opened the pull request:
I thought I'd completely tested #73 but missed out the most important bit of _actually setting_ the info dictionary items...!

Also added support for ModDate document property (docs also updated).

rkusa commented on lib/object/trailer.js:
```diff
@@ -14,10 +22,21 @@ class PDFTrailer extends PDFDictionary {
+    const infoDictionary = {
       Producer: new PDFString(info.producer),
       CreationDate: new PDFString(formatDate(info.creationDate || new Date)),
-    }))
```
Just a small nit, do you mind removing the `Producer` so that this value is added as all other properties below?

rhodgkins commented:
I've moved the default values for the document properties into the `PDFTrailer` as it seems to make more sense setting them in the same place.

I've also changed all the document properties to use that same method as you requested above (converting all the values to strings first where needed).

rkusa commented:
Thanks a lot! 👍
